### ISSUE TITLE
Misaligned indent in curried arrow function

### ISSUE THREAD

**Issue by spenceryue** [prettier/prettier#12597]
**Prettier 2.6.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwA6UAEn1ezjffSKAZxkwA9MBeTAQQDoAjACkKP3b0+NszQgAhszAATOADMA5gAsAlgCsA1gBsAtlAgAHAI4AncgFcAbgHdKATwBeI8VLlK1mnQePmrt0RMmCANBy82GL8gnY+jioaWnqGMKYWNuEOClEuse6JXva+IIGcAJS0AHz5RKxwRTTFmAAMZZgFANz5AL4c7VCdIH4gOjDy0KTIoEL6+hBmAApjCMMoQqpmQpbDvcz6QmDKcDAAykLqcAAy8lBwyJKLpHDrm9u7e9pbZ9LIMPpGtyBw6sxwYgkYmOQig0iMQmkcAAYhB9OohDABmDkMIjDAID0QLIYOpVAB1BTwUjPMBwPZzeQDExUyyosCkNYgM43fQwKabaQIy7Xb6KUiUPavVRwACKRgg8B5qhuvWehjg+lRMEs2jgpDA+nk2hgWO0Wtg+PkYhgsmQAEYAEy1OUTG74zbaVH69WKkwXXq6CXwDk6ebCUgAWnOAIBWP0cC98gjHMh3KQVxl3xu6nk70+yeFYu9FwTvN6MBERpNZqQloLm3kqleAGEIOp4z9SABWLFGG4AFRE80TspAJi+AEkoBJYHtNdqYPQR3sVSLpTdWq0gA)

```sh
--parser typescript
--print-width 120
```

**Input:**

```tsx
{
  {
    {
      const x = A.b(
        (
          c = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdef",
          d = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdef"
        ) =>
        (e) => 0
      );
    }
  }
}

```

**Output:**

```tsx
{
  {
    {
      const x = A.b(
        (
            c = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdef",
            d = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdef"
          ) =>
          (e) =>
            0
      );
    }
  }
}

```

**Expected behavior:**
Same as Input.

**Comment by GlebDolzhikov** [comment]
https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKAhgAgLyYIwCUOAfJqkdqQAwDcAOlAyADQgQAOMAltAM7Kh0AJyEQA7gAVhCfinQAbMegCe-VgCMh6MAGs4MAMroAtnAAyXKHGQAzBbzgatu-QfbbLAc2QwhAV0cQOGN1OAATMPCzdChPP3RPOAAxCCFjdBhuWOQQdD8YCBYQAAsYY3kAdWKueF53MDgDGRquADca5RywXjUQSwchGAktT3Tbe0CAK14ADwMveTgART8IeHH5B1Z3IQGcmGV2OF4wIS5OIvYz2AquMJhi5AAOKm3RBwqtdhyr47ghVrWVgAR1W8GGHFkuV4AForOFwkUhHBQVxkcMEmMkHZNoEHMYuD5-HiFsswdZsRNWDB0OpbvdHkgAEzUrRceReADCEGMWKCvAArEU-A4ACq02Q4rYgVoBACSUEisAMp3OMAAgoqDAdFhsHABffVAA let's discuss, should we move returned function to next row in this case?

**Comment by thorn0** [comment]
Fixed by #13391
